operator/controllers/controller/reconciler.go

```diff
@@ -366,10 +382,19 @@ func (r *ControllerReconciler) manageResources(ctx context.Context, log logr.Log
 	return nil
 }
 
+func (r *ControllerReconciler) createNameForConfigMap(instance *controllerv1alpha1.Controller) string {
+	if !r.MultipleControllers {
+		return controllers.ControllerServiceName
+	}
+
+	return fmt.Sprintf("controller-%s", instance.GetName())
+}
+
 // reconcileConfigMap prepares the desired states for Controller configmaps and
 // sends an request to Kubernetes API to move the actual state to the prepared desired state.
 func (r *ControllerReconciler) reconcileConfigMap(ctx context.Context, instance *controllerv1alpha1.Controller) error {
-	configMap, err := configMapForControllerConfig(instance.DeepCopy(), r.Scheme)
+	name := r.createNameForConfigMap(instance)
```

Comment:
With the flag, this should be fine as we are using the older names here.

The second option is, we can handle the migration to new names here as below:
1. Use the names based on the instance
2. In each of the reconcile functions like `reconcileConfigMap`, we can first create/update the new resources based on the names from first point.
3. We can then do a check if the Kubernetes resources with previous name `aperture-controller` exists.
4. If it exists, we can delete them
5. Use status to set that the clean-up of older resources is done so we don't repeat points 3-4 on each reconcilation loop.
6. We can then add a deprecation message that this is only supported in next 3 releases (like we do for other areas) and then remove this clean-up logic.

User can still change this field as a trial but we can assume that they won't dig that dipper while upgrading and merge this PR for now. We can handle the clean-up part in a follow-up PR

Reply:
Instead of doing that on per-resource basis, I've added deleteSingletonResources function that handles all of them at the same time. I've not added the spec change, but the reconciliation loop shouldn't run too often for that to become a problem?

Reply:
Yes we have handled the reconcile loop so that it is not triggered too frequently on start-up (it hits 4-5 calls in a second otherwise) so it should be fine